packages/langium/src/generator/generator-node.ts

```diff
@@ -4,31 +4,83 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-const EOL = (typeof process === 'undefined') ? '\n' : (process.platform === 'win32') ? '\r\n' : '\n';
+import { processGeneratorNode } from './node-processor';
 
-export type GeneratorNode = CompositeGeneratorNode | IndentNode | NewLineNode | string;
+export const EOL = (typeof process === 'undefined') ? '\n' : (process.platform === 'win32') ? '\r\n' : '\n';
+
+export type Generated = GeneratorNode | string | undefined;
+export type GeneratorNode = CompositeGeneratorNode | IndentNode | NewLineNode;
+
+export function isGeneratorNode(node: unknown): node is CompositeGeneratorNode | IndentNode | NewLineNode {
+    return node instanceof CompositeGeneratorNode
+        || node instanceof IndentNode
+        || node instanceof NewLineNode;
+}
+
+export function isNewLineNode(node: unknown): node is NewLineNode {
+    return node instanceof NewLineNode;
+}
+
+export function toString(object: unknown): string {
+    if (isGeneratorNode(object))
+        return processGeneratorNode(object);
+    else
+        return String(object);
+}
 
 export class CompositeGeneratorNode {
 
-    readonly contents: GeneratorNode[] = [];
+    readonly contents: Array<(GeneratorNode | string)> = [];
 
-    constructor(...contents: GeneratorNode[]) {
+    constructor(...contents: Generated[]) {
         this.append(...contents);
     }
 
-    append(...args: Array<GeneratorNode | ((node: CompositeGeneratorNode) => void)>): CompositeGeneratorNode {
+    isEmpty(): boolean {
+        return this.contents.length === 0;
+    }
+
+    append(...args: Array<Generated | ((node: CompositeGeneratorNode) => void)>): this {
         for (const arg of args) {
             if (typeof arg === 'function') {
                 arg(this);
-            } else {
+            } else if (arg) {
                 this.contents.push(arg);
             }
         }
         return this;
     }
 
-    indent(func?: (indentNode: IndentNode) => void): CompositeGeneratorNode {
-        const node = new IndentNode();
+    appendIf(condition: boolean, args: Generated | ((node: CompositeGeneratorNode) => void)): this {
```

Comment:
What is the use case of this method?

Reply:
It's quiet beneficial, enables conditional appending in chained method calls, like

![image](https://user-images.githubusercontent.com/8513883/207811365-d8a231f5-b3d7-4f05-b8f6-c4c855dd2222.png)